server/cp/ultrad.py

```diff
@@ -0,0 +1,135 @@
+
+import logging
+import requests
+
+from flask import current_app as app
+from urllib.parse import urljoin
+
+from apps.tasks import send_to
+from superdesk import get_resource_service
+from superdesk.lock import lock, unlock, touch
+from superdesk.text_utils import get_text
+from superdesk.celery_app import celery
+from superdesk.editor_utils import Editor3Content
+
+
+logger = logging.getLogger(__name__)
+
+ULTRAD_ID = 'ultrad_id'
+ULTRAD_URL = 'https://pc-trad.herokuapp.com/cms/'
+ULTRAD_TIMEOUT = (5, 10)
+
+
+class UltradException(RuntimeError):
+    pass
+
+
+def get_headers():
+    return {'x-ultrad-auth': app.config['ULTRAD_AUTH']}
+
+
+def upload_document(item):
+    payload = {
+        'lang': {
+            'fromLang': 'en',
+            'toLang': 'fr',
+        },
+        'name': item['headline'],
+        'state': 'new',
+        'text': {
+            'original': get_text(item['body_html']),
+        },
+    }
+
+    try:
+        resp = requests.post(ULTRAD_URL, json=payload, headers=get_headers(), timeout=ULTRAD_TIMEOUT)
+        resp.raise_for_status()
+    except requests.HTTPError:
+        logger.exception('got error on post %s', item['guid'])
+        raise UltradException()
+
+    data = get_json(resp)
+    return data['_id']
+
+
+def get_document(ultrad_id):
+    url = urljoin(ULTRAD_URL, ultrad_id)
+    try:
+        resp = requests.get(url, headers=get_headers(), timeout=ULTRAD_TIMEOUT)
+        resp.raise_for_status()
+    except requests.HTTPError:
+        logger.exception('got error on get %s', ultrad_id)
+        raise UltradException()
+    return get_json(resp)
+
+
+def get_json(resp):
+    try:
+        return resp.json()
+    except ValueError:
+        logger.error('error when parsing ultrad response "%s"', resp.text)
+        raise UltradException()
+
+
+@celery.task(soft_time_limit=300)
+def sync():
+    lock_name = 'ultrad'
+    if not lock(lock_name):
+        logger.info('lock taken %s', lock_name)
+        return
+    try:
+        desk = get_resource_service('desks').find_one(req=None, name=app.config['ULTRAD_DESK'])
+        if not desk:
+            logger.warn('ultrad desk not found name=%s', app.config['ULTRAD_DESK'])
```

Comment:
`warn` should not be used anymore, it's deprecated, please use `warning` instead: https://docs.python.org/3/library/logging.html#logging.Logger.warning

(reporting once, but it's valid for other `warn` uses below).